koopovereenkomst-v3-simple/pages/koper.tsx

```diff
@@ -1,18 +1,104 @@
-import Head from 'next/head';
-import KadasterKnowledgeGraph from '../components/kkg';
-import Layout from '../components/layout';
+import Box from "@mui/material/Box";
+import Step from "@mui/material/Step";
+import StepLabel from "@mui/material/StepLabel";
+import Stepper from "@mui/material/Stepper";
+import Typography from "@mui/material/Typography";
+import Head from "next/head";
+import * as React from "react";
+import Layout from "../src/Layout";
+import Step1 from "./steps_koper/Step1";
+import Step2 from "./steps_koper/Step2";
+
+const steps = [
+  "Datapod",
+  "Deelnemen Koop",
+  "Datapod toegang",
+  "Persoonsgegevens",
+  "Controleren",
+  "Tekenen",
+];
 
 export default function Koper() {
+  const [activeStep, setActiveStep] = React.useState(0);
+  const [skipped, setSkipped] = React.useState(new Set<number>());
+
+  const isStepOptional = (step: number) => {
+    // return step === 1;
+    return false;
+  };
+
+  const isStepSkipped = (step: number) => {
+    return skipped.has(step);
+  };
+
+  const handleNext = () => {
+    let newSkipped = skipped;
+    if (isStepSkipped(activeStep)) {
+      newSkipped = new Set(newSkipped.values());
+      newSkipped.delete(activeStep);
+    }
+
+    setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    setSkipped(newSkipped);
+  };
+
+  function ActiveStep(props) {
+    switch (props.value) {
+      case 0:
+        return <Step1 handleNext={handleNext} />;
+      case 1:
+        return <Step2 />;
+      // case 2:
+      //   return <Step3 />;
+      // case 3:
+      //   return <Step4 />;
+      // case 4:
+      //   return <Step5 />;
+      // case 5:
+      //   return <Step6 />;
+      default:
+        return <Step1 handleNext={handleNext} />;
+    }
+  }
 
-    const title = "Koper Homepage";
-
-    return (
-        <Layout role="koper">
-            <Head>
-                <title>{title}</title>
-            </Head>
-            <h1>{title}</h1>
-            <p>Content ...</p>
-        </Layout>
-    );
+  return (
+    <Layout role="koper">
+      <Head>
+        <title>{"Koper"}</title>
+      </Head>
+      <Box
+        sx={{
+          width: "100%",
+          my: 4,
+          display: "flex",
+          flexDirection: "column",
+          justifyContent: "center",
+          alignItems: "center",
+        }}
+      >
+        <ActiveStep value={activeStep} />
+        <Stepper sx={{ width: "100%", minHeight: "4rem" }} activeStep={activeStep}>
```

Comment:
Zou misschien nog wel in een über-class kunnen ... maar ik vind copy-paste hier niet zo'n probleem. Jij?

Reply:
Je bedoelt copy-paste tussen verkoper en koper? Had me er nog niet aan gestoord, maar dat komt ook doordat we alleen nog aan de verkoper kant hebben gewerkt 😄 Wat mij betreft focussen we eerst op verkoper, zodra koper aan de beurt is kunnen we dit eventueel aanpakken